sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -451,6 +458,12 @@ case class IntegralDivide(left: Expression, right: Expression) extends DivModLik
   override def evalOperation(left: Any, right: Any): Any = div(left, right)
 }
 
+object IntegralDivide {
+  def apply(left: Expression, right: Expression): IntegralDivide = {
+    new IntegralDivide(left, right)
+  }
+}
```

Comment:
shall we define `unapply` as well? Most of the time we don't care about the `returnLong` paramter. When we do, we should write `case e @ IntegralDivide(left, right) if e.returnLong`.

Reply:
There is only one place where we unapply `IntegralDivide`. It is here https://github.com/apache/spark/pull/27628/files#diff-8e1575bb706d6f7e8b5ea0b175eaeafcR178 . And `returnLong` is not checked, it is just extracted and copy-pasted.

Reply:
If I define `unapply`, it will be no used. Not sure that is is needed.